h/search/persistence.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+"""Code for persisting (indexing) annotations to Elasticsearch."""
+
+from __future__ import unicode_literals
+
+import elasticsearch_dsl
+
+
+class Annotation(elasticsearch_dsl.DocType):
+    """An elasticsearch_dsl annotation document."""
+
+    class Meta:
+        index = "hypothesis"
+
+    authority = elasticsearch_dsl.Keyword(index=False)
```

Comment:
You can view the mapping dict that this `Annotation` class generates (equivalent of our [`ANNOTATION_MAPPING`](https://github.com/hypothesis/h/blob/887b9ef7c728e662107de136410c2c534adbef68/h/search/config.py#L21-L106) for Es 1):

```python
>>> persistence.Annotation._doc_type.mapping.to_dict()
{'doc': {'properties': {'authority': {'index': False, 'type': 'keyword'}}}}
```

I _think_ `"index": False` is the equivalent of `"not_analyzed"` that we're using in Es 1.

We use `"type": "string"` in Es 1 but string no longer exists in Es 6, it's "keyword" or "text" (of which keyword seems most appropriate for the authority field)

Reply:
> I think "index": False is the equivalent of "not_analyzed" that we're using in Es 1.

They have different meanings AIUI. `"index": False` means that the field is not indexed and [cannot be queried](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index.html), `not_analyzed` means that the field's content is indexed as a single term with no pre-processing and can only be searched exactly as provided at index time. See https://www.elastic.co/guide/en/elasticsearch/reference/2.3/mapping-index.html

Reply:
Hmm. But those are the old docs. The new docs have only true and false: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index.html

Reply:
Need to look into this, not sure what the modern equivalent of not_analyzed is